Makefile

```diff
@@ -34,7 +34,56 @@ endif
 GO=go
 GOFLAGS=""
 
+# requires:
+#	AWS_SECRET_ACCESS_KEY		- Secret from AWS
+#	AWS_ACCESS_KEY_ID		- Access key from AWS
+#	S3_REGION			- Region of the bucket used to store the model
+#	S3_ENDPOINT			- Endpint of the bucket
+#	IMAGE_REGISTRY_USERNAME		- quay.io username
+#	IMAGE_REGISTRY_PASSWORD		- quay.io password
+#	IMAGE_REGISTRY_CREDENTIALS	- quay.io robot secret file
+go-test-setup:
+ifndef AWS_SECRET_ACCESS_KEY
+	$(error AWS_SECRET_ACCESS_KEY is undefined)
+endif
+ifndef AWS_ACCESS_KEY_ID
+	$(error AWS_ACCESS_KEY_ID is undefined)
+endif
+ifndef S3_REGION
+	$(error S3_REGION is undefined)
+endif
+ifndef S3_ENDPOINT
+	$(error S3_ENDPOINT is undefined)
+endif
+ifndef IMAGE_REGISTRY_USERNAME
+	$(error IMAGE_REGISTRY_USERNAME is undefined)
+endif
+ifndef IMAGE_REGISTRY_PASSWORD
+	$(error IMAGE_REGISTRY_PASSWORD is undefined)
+endif
+ifndef IMAGE_REGISTRY_CREDENTIALS
+	$(error IMAGE_REGISTRY_CREDENTIALS is undefined)
+endif
+	oc new-project test-namespace
+	@sed -e "s#{{ AWS_SECRET_ACCESS_KEY }}#${AWS_SECRET_ACCESS_KEY}#g" -e "s#{{ AWS_ACCESS_KEY_ID }}#${AWS_ACCESS_KEY_ID}#g" -e "s#{{ S3_REGION }}#${S3_REGION}#g" -e "s#{{ S3_ENDPOINT }}#${S3_ENDPOINT}#g" pipelines/tekton/aiedge-e2e/templates/credentials-s3.secret.yaml.template | oc create -f -
+	@sed -e "s#{{ IMAGE_REGISTRY_USERNAME }}#${IMAGE_REGISTRY_USERNAME}#g" -e "s#{{ IMAGE_REGISTRY_PASSWORD }}#${IMAGE_REGISTRY_PASSWORD}#g" pipelines/tekton/aiedge-e2e/templates/credentials-image-registry.secret.yaml.template | oc create -f -
+	oc create -f ${IMAGE_REGISTRY_CREDENTIALS}
+	oc secret link pipeline $(shell yq '.metadata.name' ${IMAGE_REGISTRY_CREDENTIALS})
+
+# requires:
+#	S3_BUCKET		- Name of S3 bucket that has the model
+#	TARGET_IMAGE_REPO	- Image repository that built model will be pushed to
+go-test: go-test-setup
+ifndef S3_BUCKET
+	$(error S3_BUCKET is undefined)
+endif
+ifndef TARGET_IMAGE_REPO
+	$(error TARGET_IMAGE_REPO is undefined)
+endif
+	oc project test-namespace
+	(cd test/e2e-tests/tests && S3_BUCKET=${S3_BUCKET} TARGET_IMAGE_REPO=${TARGET_IMAGE_REPO} ${GO} test -timeout 30m)
+	oc delete project test-namespace
```

Comment:
I'm not suggesting that this needs to be changed, just highlighting that it's a little asymmetrical (for lack of a better term) that this target doesn't create the namespace, but it does delete it.

Trying to understand @LaVLaS [comment](https://github.com/opendatahub-io/ai-edge/pull/188#discussion_r1478552193), should test setup and teardown should be decoupled from `go-test`? In essence, if I run `go-test-setup` once, should I be able to run `go-test` many times, and then eventually delete the namespace when I'm done?

If `go-test-setup` is required before _each_ invocation of `go-test`, then maybe there should only be `go-test`, and it should incorporate the existing recipe that's in `go-test-setup`?

Reply:
This is what I was also confused about aswell when it was suggested to seperate them. Because the setup requires a namespace to be present as we are interacting with the cluster I feel like it is then a part of the tests aswell.

Reply:
One thing maybe is that neither the test or test-setup targets delete or create the namespace and always use the current one in use

Reply:
In that case, my suggestion for now, would be to not delete the namespace as part of this recipe (or change to a static `test-namespace` directory), and not have the `go-test-setup` one as a prerequisite. That would mean that someone could do something along the lines of the following:

```bash
make go-test-setup
make go-test
make go-test
...
```

Reply:
Okay I get you, I think that shouldn't be that much of a change either

Reply:
@LaVLaS was this what you were expecting, just want to make sure this is the setup we want

Reply:
Done 👍